### ISSUE TITLE
Research and decide on app infrastructure

### ISSUE THREAD

**Issue by ktohalloran** [azavea/raster-vision-marketing-site#1]
There are 3 important decisions that need to be made as a part of this task:
- Decide on a framework, such as NextJS, Jekyll, Gatsby, etc.
- Decide on a UI library, such as Tailwind CSS, Chakra UI, etc.
- Decide whether we would like to add Docker and use any STRTA

This task should include finding and researching examples of other projects that use these tools to determine which would be the best fit.

**Comment by ktohalloran** [comment]
## TL;DR

Due to the size and complexity of the site and in the interest of minimizing training and learning time, this project will be built without a framework or static site generator using hand-written HTML, CSS, Javascript, jQuery, and Chakra UI. It will not use Docker or any STRTA and instead use yarn and yvm directly on host for dependency management.

## Overview

This site is a one-page, static site that includes a variety of visualizations, including static images and animations. As such, there were 3 decisions to be made: 
- Which framework and static site generator
- Which UI library or framework
- Whether to containerize and automate development and deployment processes

**Framework**

I compared 3 different static site generators:
- NextJS
- Gatsby
- Jekyll

None of the 3 of these generators quite met our needs; as a one-page site, it was not a great fit for Jekyll, and between Gatsby and NextJS, the latter felt like the closest fit, as we have a couple of working examples; however, it was also unnecessarily heavy duty for this use case. While I was drawn by NextJS’s support for React, which felt closer to being within my wheelhouse, I ultimately decided to not use a framework at all due to the site being one page with minimal complexity. As a result, I will also not use any static site generator and will hand-write the app using HTML, CSS, Javascript, and jQuery.

**UI Library/Framework**

I compared 3 different UI frameworks/libraries:
- Tailwind CSS
- Chakra UI
- Material UI

I also briefly considered Theme UI but research suggested Chakra is “heavily inspired by” Theme UI and was therefore a more robust alternative. Although the designs were made with Tailwind in mind and the highly customizable nature of Tailwind was a strong benefit, I ultimately rejected it because it would require either constructing our own components or paying for the right to use theirs. While I do think that it would be beneficial to explore Tailwind at some point, in the interest of keeping down training and learning time, I’ve opted for a component library instead. Material UI and Chakra UI have many of the same benefits; we have working examples, have a large and active community, and include a variety of pre-built components. The differences appear in scalability, number of components, and flexibility and responsiveness of styling; as a one page app, the most important feature for this project was the flexibility of styling, which Chakra has over Material. This, combined with the fact that Chakra has been codified as our default UI component library in our [app template](https://github.com/azavea/azavea-project-template), settled the matter.

**Containerization/Automation**

Due to the size of the site and expected number of dependencies, I do not intend to use Docker or add any STRTA. Instead, I plan to handle package management via yarn and yvm, as this is the chosen package manager of our [app template](https://github.com/azavea/azavea-project-template). 

Thanks to @BryanQuigley and @ddohler for their guidance and suggestions!
